src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -153,6 +158,12 @@ export class MockHttpSocket extends MockSocket {
     }
 
     const socket = this.createConnection()
+
+    // If the developer destroys the socket, destroy the original connection.
+    this.once('error', (error) => {
```

Comment:
Added this: if the socket gets destroyed by the developer, abort the pending actual socket connection.